geppetto.js/geppetto-client/js/common/middleware/geppettoMiddleware.js

```diff
@@ -0,0 +1,83 @@
+import { clientActions } from '../actions';
+import EventManager, { callbacksList } from '@metacell/geppetto-meta-client/common/EventManager';
+
+
+export function callbacksMiddleware ({ getState, dispatch }) {
+  return function (next) {
+    return function (action) {
+      
+      console.debug(action);
+      var actionTriggered = false;
+      if (callbacksList[action.type] !== undefined && callbacksList[action.type].size > 0) {
+        callbacksList[action.type].forEach(item => {
+          item(action);
+        });
+      }
+
+      switch (action.type) {
+      case clientActions.SELECT:
+        // FIXME: do we really need the action focus changed? this can be handled directly by the SELECT action itself
+        EventManager.actionsHandler[EventManager.clientActions.FOCUS_CHANGED](action.data.scope);
+        // FIXME: do we need to use the widgetsListener?
+        GEPPETTO.WidgetsListener.update(GEPPETTO.Events.Select);
+        break;
+      case clientActions.DO_EXPERIMENT_PLAY:
+        Project.getActiveExperiment().playAll();
+        break;
+      case clientActions.EXPERIMENT_ACTIVE:
+        GEPPETTO.WidgetsListener.update(GEPPETTO.WidgetsListener.WIDGET_EVENT_TYPE.DELETE);
+        break;
+      case clientActions.EXPERIMENT_PLAY:
+        GEPPETTO.WidgetsListener.update(GEPPETTO.Events.Experiment_play, action.data.parameters);
+        break;
+      case clientActions.EXPERIMENT_LOADED:
+        // From the ProjectController
+        GEPPETTO.ProjectsController.refreshUserProjects();
+        // From the GEPPETTO.Events
+        if (GEPPETTO.UserController.isLoggedIn()) {
+          EventManager.actionsHandler[EventManager.clientActions.HIDE_SPINNER]();
+        }
+        break;
+      case clientActions.EXPERIMENT_OVER:
+        GEPPETTO.WidgetsListener.update(GEPPETTO.Events.Experiment_over);
+        var id = action.data.experiment.id;
+        var name = action.data.experiment.name;
+        if (GEPPETTO.ExperimentsController.playLoop === true) {
+          Project.getActiveExperiment().play({ step: 1 });
+        } else {
+          GEPPETTO.CommandController.log("Experiment " + name + " with " + id + " is over ");
+        }
+        break;
+      case clientActions.EXPERIMENT_UPDATE:
+        /*
+         * This fix the tons of experiment_update action updates we trigger durint the experiment.
+         *
+         * if (this.getState().client.experiment.status === clientActions.EXPERIMENT_UPDATE) {
```

Comment:
Confused by this, why is the fix disabled? Can we remove this comment?

Reply:
Probably we don't even need the experiment workflows to be in place here